resource-validator/src/main/scala/com/broadinstitute/dsp/resourceValidator/DeletedRuntimeChecker.scala

```diff
@@ -37,28 +37,42 @@ object DeletedRuntimeChecker {
         implicit ev: Ask[F, TraceId]
       ): F[Option[Runtime]] =
         for {
-          clusterOpt <- deps.dataprocService
+          clusterOpt <- deps.runtimeCheckerDeps.dataprocService
             .getCluster(runtime.googleProject, regionName, DataprocClusterName(runtime.runtimeName))
-          _ <- clusterOpt.traverse_ { _ =>
-            if (isDryRun)
-              logger.warn(s"${runtime} still exists in Google. It needs to be deleted.")
-            else
-              logger.warn(s"${runtime} still exists in Google. Going to delete it.") >> deps.dataprocService
-                .deleteCluster(runtime.googleProject, regionName, DataprocClusterName(runtime.runtimeName))
-                .void
+          r <- clusterOpt.flatTraverse { _ =>
+            for {
+              isBillingEnabled <- deps.billingService.isBillingEnabled(runtime.googleProject)
+              r <- if (isDryRun)
+                logger
+                  .warn(
+                    s"$runtime still exists in Google. It needs to be deleted. isBillingEnabled: $isBillingEnabled. Project: ${runtime.googleProject}"
+                  )
+                  .as(Option(runtime))
+              else
+                isBillingEnabled match {
+                  case true =>
+                    logger.warn(s"$runtime still exists in Google and billing is enabled. Going to delete it.") >> deps.runtimeCheckerDeps.dataprocService
+                      .deleteCluster(runtime.googleProject, regionName, DataprocClusterName(runtime.runtimeName))
+                      .as(Option(runtime))
+                  case false =>
+                    logger
+                      .warn(s"$runtime still exists with an anomaly, but cannot delete it because billing is disabled.")
```

Comment:
shall we log it at `info` level, and just say something like "$runtime doesn't exist in google anymore because billing is disable for $project"

Reply:
It does exist in google though, we just cannot delete it because billing is disabled.

Reply:
my guess is the cluster doesn't actually exist in google (it's just getCluster call returns a Some) because for it to exist, it'll incur cost to google, and google can't charge for it

Reply:
i changed it to `$runtime has been reported from getCluster, but billing is disabled so cannot perform any actions`